harness/determined/pytorch/_pytorch_context.py

```diff
@@ -168,25 +174,39 @@ def wrap_model(self, model: torch.nn.Module) -> torch.nn.Module:
             check.false(self._use_apex, "Must call wrap_model() before configure_apex_amp.")
 
             model = model.to(self.device)
-            if not self.distributed.size > 1 and self.n_gpus > 1:
-                check.eq(
-                    self._aggregation_frequency,
-                    1,
-                    "Please enable `optimized_parallel` to use aggregation "
-                    "frequency greater than 1 for single machine multi-GPU "
-                    "training.",
-                )
-                model = nn.DataParallel(model)
-                logging.debug("Initialized model for native parallel training.")
+
+            if self.distributed.size > 1 and self._distributed_backend.use_torch():
+
+                class PyTorchDistributedDataParallel(
+                    torch.nn.parallel.DistributedDataParallel  # type: ignore
+                ):
+                    """
+                    Pass-through Model Wrapper to enable access to inner module attributes
+                    when using PyTorch DDP
+                    """
+
+                    def __getattr__(self, name: str) -> Any:
+                        try:
+                            return super().__getattr__(name)
+                        except AttributeError:
+                            return getattr(self.module, name)
+
+                wrapped_model = PyTorchDistributedDataParallel(model)
```

Comment:
can you add a comment here about why we don't need to specify device_ids and output_device when initializing DDP module?  
cf. https://github.com/NVIDIA/apex/blob/master/examples/imagenet/README.md
cf. https://github.com/pytorch/pytorch/blob/master/torch/nn/parallel/distributed.py#L428